### DISCUSSION TITLE
How to stop the script if quality gate in SonarQube fails

### DISCUSSION THREAD

**Discussion by umairjadoon** [cake-build/cake#3014] [Q&A] [answered]
In CI pipeline, i am running a cake script from Jenkinsfile to build, analyze, compile and test the application. Can anyone please help me in figuring out how can i make sure that if Quality gate in SonarQube fails, my build also fails.

**Comment by pascalberger** [comment, marked as answer]
Cake itself doesn't doesn't ship with support for SonarQube. There's a [Cake.SonarResults](https://cakebuild.net/extensions/cake-sonarresults/) 3rd party addin which seems to provide an alias to break a build based on SonarQube quality gate. If you don't want to use a 3rd party addin you can make the call to the SonarQube API directly. See SonarQube documentation for details how to do this.
